daemon/list.go

```diff
@@ -28,13 +30,15 @@ type ContainersConfig struct {
 
 func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container, error) {
 	var (
-		foundBefore bool
-		displayed   int
-		all         = config.All
-		n           = config.Limit
-		psFilters   filters.Args
-		filtExited  []int
+		foundBefore    bool
+		displayed      int
+		ancestorFilter bool
+		all            = config.All
+		n              = config.Limit
+		psFilters      filters.Args
+		filtExited     []int
 	)
+	imagesFilter := map[string]struct{}{}
```

Comment:
if you use `map[string]bool` you can simplify the conditionals to something like `if imagesFilter[imageID] {` and remove those extra `ok`